### PR TITLE
Add missing derived lens documentation

### PR DESCRIPTION
**Pull request by lidin**
If you enable `warn(missing_docs)` lint when deriving `Lens` on a type, it will detect missing documentation on the `#ty_derived_lenses` module, as well as the `fn new()` associated function on the lenses. This PR fixes that by adding the missing documentation.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ You can find its changes [documented below](#070---2021-01-01).
 - `Notification`s can be submitted while handling other `Notification`s ([#1640] by [@cmyr])
 - Added ListIter implementations for OrdMap ([#1641] by [@Lejero])
 - `Padding` can now use `Key<Insets>` ([#1662] by [@cmyr])
+- Added missing documentation on derived lens items ([#1696] by [@lidin])
```
Comment (SecondFlight) [inline CHANGELOG.md:33]: You have to add links to the section at the bottom of the file for these to work.

Comment (lidin) [reply CHANGELOG.md:33]: Done!

**Comment by SecondFlight** [comment]
Thanks for addressing the comment! I'll have to let someone else approve this PR. I don't know quite enough about derived lenses to know if the wording is accurate.

File: druid-derive/src/lens.rs [hunk 2]
```diff
@@ -83,18 +83,24 @@ fn derive_struct(input: &syn::DeriveInput) -> Result<proc_macro2::TokenStream, s
...
-        let docs = format!(
-            "Lens for the field `{}` on [`{1}`](super::{1})",
+        let struct_docs = format!(
+            "Lens for the field `{}` on [`{1}`](super::{1}).",
+            field_name, ty
+        );
+
+        let fn_docs = format!(
+            "Creates a new lens for the field `{}` on [`{1}`](super::{1}). Use [`{1}::{0}`](super::{1}::{0}) instead.",
             field_name, ty
```
Comment (Maan2003) [inline druid-derive/src/lens.rs:93]: can you name the format arguments, (for some reason, I am not able to suggest changes on two lines :thinking:)

Comment (lidin) [reply druid-derive/src/lens.rs]: Sorry, I misinterpreted your feedback! I've pushed a fix. How's it looking now?

**Review by Maan2003 — APPROVED** [review]
Thanks!

**Pull request merged**